src/modules/commander/commander.cpp

```diff
@@ -855,13 +855,22 @@ bool handle_command(struct vehicle_status_s *status_local, const struct safety_s
 					status.arming_state = vehicle_status_s::ARMING_STATE_IN_AIR_RESTORE;
 
 				} else {
+
 					// Refuse to arm if preflight checks have failed
 					if ((!status_local->hil_state) != vehicle_status_s::HIL_STATE_ON && !status_flags.condition_system_sensors_initialized) {
 						mavlink_log_critical(&mavlink_log_pub, "Arming DENIED. Preflight checks have failed.");
 						cmd_result = vehicle_command_s::VEHICLE_CMD_RESULT_DENIED;
 						break;
 					}
 
+					// Make sure we do not takeoff as a fixed wing when the vehicle type is VTOL
+					if ( status_local->is_vtol && !status_local->is_rotary_wing )
```

Comment:
The preflight_check() code does not run during HITL. We are doing quite a lot of development in HITL, and would really like to have this check in place (makes it more representative of the real-world situation)